### ISSUE TITLE
jpm vs what you are doing

### ISSUE THREAD

**Issue by tofutim** [Noitidart/ostypes_playground#1]
Are you doing something other than the add-on SDK? I am using jpm.

**Comment by Noitidart** [comment]
Ah this is using without SDK.

However it is the same. You can just copy all the stuff from bootstrap.js into your index.js :)

I dont have the jpm/nodejs stuff setup.

**Comment by Noitidart** [comment]
So all the ctypes is correct. I updated ostypes to have the constant and the delete timer function:

https://github.com/Noitidart/ostypes_playground/tree/f001165ec47cc487e4590404dbcc5f04b2bbc49d

However it does crash. We have to keep digging at why it is, but this is definitely on the right track.

**Comment by tofutim** [comment]
Ahh, so you see it too. I actually came to this function because I was using the callback for ReadFileEx and finding that it would only callback after I closed Firefox. However, it did not crash like this - although maybe my function was persisting.

**Comment by Noitidart** [comment]
Here is how the winapi docus use this function:

https://msdn.microsoft.com/en-us/library/windows/desktop/ms687003%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396

They create their own timer queue with `CreateTimerQueue` - if you do that it should work without crashing. I always have to post on stackoverflow asking how the winapi works. I'm not an expert on that part. But once we understand the API we can get it to work without issue. The GC of the callback is no longer the issue.

**Comment by Noitidart** [comment]
I think because we did not call `CreateTimerQueue` I don't think we can use `WT_EXECUTEDEFAULT` we have to probably use one of the other flags.

**Comment by tofutim** [comment]
I think you are right about the CreateTimerQueue, though the docs suggest
that if you don't do this, it will use the default queue....

**Comment by Noitidart** [comment]
Yeah I agree it should have worked. It's confusing. If you really have to use it, I would ask on stackoverflow in the winapi tag. They have been very helpful to me.
